slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -329,6 +325,120 @@ void SLAMServiceImpl::SetUpMapBuilder() {
 }
 
 std::string SLAMServiceImpl::GetLatestJpegMapString(bool add_pose_marker) {
+    std::unique_ptr<cartographer::io::PaintSubmapSlicesResult> painted_slices =
+        nullptr;
+    try {
+        painted_slices =
+            std::make_unique<cartographer::io::PaintSubmapSlicesResult>(
+                GetLatestPaintedMapSlices());
+    } catch (std::exception &e) {
+        // jpeg will be empty which captures the error
+        LOG(INFO) << "Error creating jpeg map: No submaps available";
```

Comment:
[nit] I would log `"Error creating jpeg map: " << e.what()`, in case we add more exceptions in the future. Same for the other logging statement in the catch.